internal/weldr/api.go

```diff
@@ -848,6 +848,29 @@ func (api *API) blueprintsDepsolveHandler(writer http.ResponseWriter, request *h
 	})
 }
 
+// setPkgEVRA replaces the version globs in the blueprint with their EVRA values from the dependencies
+//
+// The dependencies must be pre-sorted for this function to work properly
+// It will return an error if it cannot find a package in the dependencies
+func setPkgEVRA(dependencies []rpmmd.PackageSpec, packages []blueprint.Package) error {
```

Comment:
Shouldn't we use this function also in `composeHandler`? I think lorax does freeze the blueprint before making a compose. See https://github.com/osbuild/osbuild-composer/issues/143

Reply:
I think that's future work, the way (AFAIK) osbuild works doesn't support passing the version into the pipeline, it seems to depend on the state of the metadata at the time of the depsolve.

Reply:
You can pass the versions into the pipeline, the packages provided are passed as-is to dnf. Though might make more sense to wait for the dnf/rpm rework to land.

Reply:
Right. I think that's a separate issue from this PR.